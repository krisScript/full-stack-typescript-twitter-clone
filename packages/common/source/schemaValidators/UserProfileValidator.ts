import * as yup from 'yup';

interface UserProfileValidatorType {
  username: string;
  handle: string;
  email: string;
  password: string;
  confirmPassword: string;
}
const UserProfileValidator = yup.object<UserProfileValidatorType>().shape({
  username: yup
    .string()
    .trim()
    .min(1)
    .max(50)
    .required(),
  handle: yup
    .string()
    .trim()
    .min(1)
    .max(50)
    .required(),
  website: yup
    .string()
    .trim()
    .url()
    .required(),
});

export default UserProfileValidator;
