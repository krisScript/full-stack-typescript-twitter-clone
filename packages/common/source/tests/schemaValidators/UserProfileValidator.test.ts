import UserProfileValidator from '@schemaValidators/UserProfileValidator';

describe('UserProfileValidator', (): void => {
  const username = 'username';
  const handle = 'testUserHandle';
  const website = 'https://testwebsite.test';
  it(`should validate successfully`, async (): Promise<void> => {
    const user = {
      username,
      handle,
      website,
    };
    await expect(
      UserProfileValidator.validate(user, { abortEarly: false }),
    ).resolves.toBe(user);
  });
  it(`should throw an error`, async (): Promise<void> => {
    const user = {
      username: '',
      handle: '',
      website: '',
    };
    await expect(
      UserProfileValidator.validate(user, { abortEarly: false }),
    ).rejects.toMatchSnapshot();
  });
});
