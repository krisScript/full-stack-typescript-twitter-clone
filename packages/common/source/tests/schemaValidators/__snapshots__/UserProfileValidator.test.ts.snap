// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileValidator should validate unsuccessfully 1`] = `[ValidationError: 5 errors occurred]`;
